arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -1074,16 +1075,23 @@ IResearchAnalyzerFeature::IResearchAnalyzerFeature(arangodb::application_feature
     arangodb::auth::Level const& level // access level
 ) {
   auto& ctx = arangodb::ExecContext::current();
-  return ctx.canUseDatabase(vocbase.name(), level) && // can use vocbase
-         ctx.canUseCollection(vocbase.name(), ANALYZER_COLLECTION_NAME, level); // can use analyzers
+  return ctx.hasAccess(vocbase.resource(), level) && // can use database
+    ctx.hasAccess(auth::CollectionResource{vocbase, ANALYZER_COLLECTION_NAME}, level); // can use analyzers
 }
 
 /*static*/ bool IResearchAnalyzerFeature::canUse( // check permissions
   irs::string_ref const& name, // analyzer name (already normalized)
   arangodb::auth::Level const& level // access level
 ) {
   auto& ctx = arangodb::ExecContext::current();
-  if (ctx.isAdminUser()) {
+  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+  // was: if (ctx.isAdminUser()) {
```

Comment:
@gnusi @Dronplane what is protected by this check? The check was "is admin user", but the comment says "authentication disabled".

Reply:
Looks like comment is correct. There should be 
`if (!ctx.isAuthEnabled())`
 as previously there  was :
`  auto* ctx = arangodb::ExecContext::CURRENT;`
`  if (!ctx) {`